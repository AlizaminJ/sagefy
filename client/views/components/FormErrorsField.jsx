import React from 'react'
import get from 'lodash.get'
import { shape, string } from 'prop-types'
import Icon from './Icon'

export default function FormErrorsField({ formErrors, field }) {
  if (!get(formErrors, field)) return null
  return (
    <ul className="FormErrors list-style-none">
      {get(formErrors, field).map(message => (
        <li>
          <mark>
            <Icon i="error" /> {message}
          </mark>
        </li>
      ))}
    </ul>
  )
}

FormErrorsField.propTypes = {
  formErrors: shape({}).isRequired,
  field: string.isRequired,
}
