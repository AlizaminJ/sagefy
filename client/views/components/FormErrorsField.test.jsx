import FormErrorsField from './FormErrorsField'

describe('FormErrorsField component', () => {
  it('should render a FormErrorsField null', () => {
    expect(FormErrorsField({ formErrors: {}, field: 'all' })).toMatchSnapshot()
  })

  it('should render a FormErrorsField data', () => {
    expect(
      FormErrorsField({
        formErrors: {
          all: [{ message: 'Hello' }],
        },
        field: 'all',
      })
    ).toMatchSnapshot()
  })
})
