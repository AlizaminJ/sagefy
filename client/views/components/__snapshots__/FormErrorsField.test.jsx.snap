// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormErrorsField component should render a FormErrorsField data 1`] = `
<ul
  className="FormErrors ls-n"
>
  <li>
    <mark>
      <Icon
        i="error"
        s="m"
      />

      Object {
        "message": "Hello",
      }
    </mark>
  </li>
</ul>
`;

exports[`FormErrorsField component should render a FormErrorsField null 1`] = `null`;
